packages/zoe/src/contractSupport/bondingCurves.js

```diff
@@ -32,12 +35,14 @@ export const getInputPrice = (
   outputReserve,
   feeBasisPoints = 30,
 ) => {
-  const oneMinusFeeInTenThousandths = subtract(10000, feeBasisPoints);
-  const inputWithFee = multiply(inputValue, oneMinusFeeInTenThousandths);
-  const numerator = multiply(inputWithFee, outputReserve);
-  const denominator = add(multiply(inputReserve, 10000), inputWithFee);
-
-  return floorDivide(numerator, denominator);
+  if (inputReserve <= 0 || outputReserve <= 0 || inputValue < 0) {
+    return 0;
+  }
```

Comment:
As we discussed, should be error rather than zero. Best is for these errors to come from assert failures.
```suggestion
  assert(inputReserve > 0, details`inputReserve ${inputReserve} must be positive`);
  assert(outputReserve > 0, details`outputReserve ${outputReserve} must be positive`);
  Nat(inputValue);
```